src/front-door/azext_front_door/custom.py

```diff
@@ -1098,3 +1112,301 @@ def list_custom_rule_match_conditions(cmd, resource_group_name, policy_name, rul
+    client = cf_fd_rules_engines(cmd.cli_ctx, None)
+
+    match_conditions = []
+    condition = RulesEngineMatchCondition(rules_engine_match_variable=match_variable,
+                                          rules_engine_operator=operator,
+                                          rules_engine_match_value=match_values,
+                                          selector=selector,
+                                          negate_condition=negate_condition, transforms=transforms)
+    if condition is not None:
+        match_conditions.append(condition)
+
+    request_header_actions = []
+    if action_type.lower() == 'requestheader':
+        request_header_actions.append(HeaderAction(
+            header_action_type=header_action,
+            header_name=header_name,
+            value=header_value))
+
+    response_header_actions = []
+    if action_type.lower() == 'responseheader':
+        response_header_actions.append(HeaderAction(
+            header_action_type=header_action,
+            header_name=header_name,
+            value=header_value))
+
+    action = RulesEngineAction(request_header_actions=request_header_actions,
+                               response_header_actions=response_header_actions)
+
+    rule = RulesEngineRule(name=rule_name, priority=priority,
+                           action=action, match_conditions=match_conditions,
+                           match_processing_behavior=match_processing_behavior)
+
+    rules_list = []
+    try:
+        rules_engine = client.get(resource_group_name, front_door_name, rules_engine_name)
+        rules_engine.rules.append(rule)
+        rules_list = rules_engine.rules
+    except ErrorResponseException as e:
+        if e.response.status_code == 404:
+            rules_list = [rule]
+        else:
+            # If the error isn't a 404, rethrow it.
+            raise e
+
+    return client.create_or_update(resource_group_name, front_door_name, rules_engine_name, rules_list)
+
+
+def delete_rules_engine_rule(cmd, resource_group_name, front_door_name, rules_engine_name, rule_name):
+    client = cf_fd_rules_engines(cmd.cli_ctx, None)
+    rules_engine = client.get(resource_group_name, front_door_name, rules_engine_name)
+    rules_engine.rules = [x for x in rules_engine.rules if x.name.lower() != rule_name.lower()]
+    if not rules_engine.rules:
+        from knack.util import CLIError
+        raise CLIError("Rules Engine must at least contain one rule")
+
+    return client.create_or_update(resource_group_name, front_door_name, rules_engine_name, rules=rules_engine.rules)
+
+
+def show_rules_engine_rule(cmd, resource_group_name, front_door_name, rules_engine_name, rule_name):
+    client = cf_fd_rules_engines(cmd.cli_ctx, None)
+    rules_engine = client.get(resource_group_name, front_door_name, rules_engine_name)
+    try:
+        return next(x for x in rules_engine.rules if x.name.lower() == rule_name.lower())
```

Comment:
I'm not sure whether we can return a iterator or not. Could you please add a test for the show command? or we can turn it by `list(next())`

Reply:
Added test case, it works fine and so is [this line](https://github.com/Azure/azure-cli-extensions/blob/master/src/front-door/azext_front_door/custom.py#L1028) above.